File: AbstractFactoryExampleJava/src/concretefactories/SwampFactory.java
package concretefactories;

import abstractions.Enemy;
import abstractions.Environment;
import abstractions.LevelFactory;
import constants.GlobalConstants;
import products.Hydra;
import products.MudMonster;
import products.MuddyLake;

public class SwampFactory extends LevelFactory {
    @Override
    public String[] getAvailableEnemyNames() {
        return new String[] { "Mud Monster", "Hydra" };
    }

    @Override
    public Enemy createEnemy(String enemyName) {
        return switch (enemyName) {
            case "Mud Monster" -> new MudMonster();
            case "Hydra" -> new Hydra();
            default -> throw new IllegalArgumentException(GlobalConstants.INVALID_CHOICE);
        };
    }

    @Override
    public Environment createEnvironment() {
        return new MuddyLake();
    }
}

File: AbstractFactoryExercise/AbstractFactoryExercise/ConcreteFactories/SwampFactory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AbstractFactoryExercise.Abstractions;
using AbstractFactoryExercise.Constants;
using AbstractFactoryExercise.Products;

namespace AbstractFactoryExercise.ConcreteFactories
{
    public class SwampFactory : LevelFactory
    {
        public override string[] GetAvailableEnemyNames()
        {
            return new string[]
                {
                "Mud Monster",
                "Hydra",
                };
        }
        public override Abstractions.Enemy CreateEnemy(string enemyName)
        {
            return enemyName switch
            {
                "Mud Monster" => new MudMonster(),
                "Hydra" => new Hydra(),
                _ => throw new ArgumentException(GlobalConstants.INVALID_CHOICE)
            };
        }
        public override Abstractions.Environment CreateEnvironment()
        {
            return new Products.MuddyLake();
        }
    }
}
